apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -3029,9 +3029,23 @@ impl Display for OpGraphPath {
                 Some(e) => {
                     let tail = self.graph.graph().edge_endpoints(e).unwrap().1;
                     let node = &self.graph.graph()[tail];
-                    let edge = &self.graph.graph()[e];
-                    let label = edge.transition.to_string();
-                    write!(f, " --[{label}]--> {node}")
+                    if i == 0 && head.root_kind.is_some() {
+                        write!(f, "{node}")
```

Comment:
this still adds extra `*` at the end of the subgraph transition node i.e. `Query(S2)` vs `Query(S2)*`

Reply:
seems like a bug as in JS that node is not marked as root

Reply:
we were incorrectly looking at the `root_kind` struct field  vs `is_root_node()` function value